<!-- 
This launch file starts a single hokuyo laser scanner,
prepares scan matching and runs the obstacle detector.
It also opens Rviz for data visualisation. 
-->
<launch>
<!--
  <node name="front_hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" required="false" output="screen">
    <param name="port" type="string" value="/dev/ttyACM0"/>
    <param name="frame_id" type="string" value="front_laser"/>

    <param name="intensity" type="bool" value="false"/>
    <param name="calibrate_time" type="bool" value="true"/>

    <param name="min_ang" value="-2.0862"/>
    <param name="max_ang" value="2.0923"/>
    <param name="skip" value="0"/>
    <param name="time_offset" value="0.0"/>
    <param name="cluster" value="1"/>

    <remap from="scan" to="front_scan"/>
  </node>
-->

  <node name="static_scan_publisher" pkg="obstacle_detector" type="static_scan_publisher" respawn="false" required="false" output="screen">
    <!--<remap from="scan" to="front_scan"/>-->
  </node>

  <node name="world_to_scanner" pkg="tf" type="static_transform_publisher" respawn="false" required="false" output="screen" args="0 0 0 0 0 0 world scanner 100"/>
 
<!--  
  <node name="scan_matching" pkg="laser_scan_matcher" type="laser_scan_matcher_node" respawn="false" required="false"  output="screen">
    <param name="fixed_frame" value="/world"/>
    <param name="base_frame" value="/base"/>
    
    <param name="use_imu"  value="false"/>
    <param name="use_odom" value="false"/>
    <param name="use_vel"  value="false"/>
    <param name="use_cloud_input" value="false"/>

    <param name="publish_tf" value="true"/>
    <param name="publish_pose" value="true"/>

    <param name="kf_dist_linear" value="0.0"/>
    <param name="kf_dist_angular" value="0.0"/>
  </node> 
-->

  <node name="obstacle_detector" pkg="obstacle_detector" type="obstacle_detector" respawn="false" required="false" output="screen">
    <param name="world_frame"    type="string" value="world"/>
    <param name="scanner_frame"  type="string" value="scanner"/>
    <param name="scan_topic"     type="string" value="scan"/>
    <param name="pcl_topic"      type="string" value="pcl"/>
    <param name="obstacle_topic" type="string" value="obstacles"/>
    <param name="marker_topic"   type="string" value="obstacle_markers"/>

    <param name="use_scan"            type="bool" value="true"/>
    <param name="use_pcl"             type="bool" value="false"/>
    <param name="publish_markers"     type="bool" value="true"/>
    <param name="save_snapshot"       type="bool" value="false"/>
    <param name="use_split_and_merge" type="bool" value="false"/>

    <param name="min_group_points"    value="3"/>
    <param name="max_group_distance"  value="0.100"/>
    <param name="distance_proportion" value="0.006136"/>
    <param name="max_split_distance"  value="0.100"/>

    <param name="max_merge_separation" value="0.200"/>
    <param name="max_merge_spread"     value="0.070"/>
    <param name="max_circle_radius"    value="0.300"/>
    <param name="radius_enlargement"   value="0.050"/>
  </node>
  
  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find obstacle_detector)/resources/obstacle_detector.rviz"/>
</launch>

